odoo/addons/base/models/ir_mail_server.py

```diff
@@ -275,17 +303,28 @@ def test_smtp_connection(self):
                     # ignored, just a consequence of the previous exception
                     pass
 
-        message = _("Connection Test Successful!")
+        messages = [_('Connection Test Successful!')]
+        if autodetect_max_email_size:
+            if self.max_email_size:
+                messages.append(_('Email maximum size updated to %(size)s MB',
+                                  size=', '.join(f"{server.max_email_size:.2f}" for server in self)))
+            else:
+                raise UserError(_("The server doesn't return the maximum email size."))
```

Comment:
Why not raise this here above? Looks like other server issues are thrown inside the recordset loop.